src/platforms/common/usb_serial.c

```diff
@@ -103,7 +103,11 @@ static usbd_request_return_codes_e gdb_serial_control_request(usbd_device *dev,
 	switch (req->bRequest) {
 	case USB_CDC_REQ_SET_CONTROL_LINE_STATE:
 		/* Send a notification back on the notification endpoint */
-		usb_serial_set_state(dev, req->wIndex, CDCACM_GDB_ENDPOINT + 1U);
+#if defined(STM32F4) && CDCACM_GDB_NOTIF_ENDPOINT < 4
```

Comment:
This condition needs amending to only exclude STM32F4 and `CDCACM_GDB_NOTIF_ENDPOINT >= 4` - otherwise this changes the behaviour on the F1 and other parts.

Reply:
Good catch. I'll try rewriting the macro condition and moving one set of them into `usb_serial_set_state()`. The other set is dirty because `usb_ep_setup()` is part of libopencm3 API, and I didn't feel like introducing a wrapper.